scipy/constants/__init__.py

```diff
@@ -22,36 +22,36 @@
 ==================
 
 ===========================  =================================================================
-``c``                        speed of light in vacuum
-``speed_of_light``           speed of light in vacuum
-``mu_0``                     the magnetic constant :math:`\mu_0`
-``epsilon_0``                the electric constant (vacuum permittivity), :math:`\epsilon_0`
-``h``                        the Planck constant :math:`h`
-``Planck``                   the Planck constant :math:`h`
-``hbar``                     :math:`\hbar = h/(2\pi)`
-``G``                        Newtonian constant of gravitation
-``gravitational_constant``   Newtonian constant of gravitation
-``g``                        standard acceleration of gravity
-``e``                        elementary charge
-``elementary_charge``        elementary charge
-``R``                        molar gas constant
-``gas_constant``             molar gas constant
-``alpha``                    fine-structure constant
-``fine_structure``           fine-structure constant
-``N_A``                      Avogadro constant
-``Avogadro``                 Avogadro constant
-``k``                        Boltzmann constant
-``Boltzmann``                Boltzmann constant
-``sigma``                    Stefan-Boltzmann constant :math:`\sigma`
-``Stefan_Boltzmann``         Stefan-Boltzmann constant :math:`\sigma`
-``Wien``                     Wien displacement law constant
-``Rydberg``                  Rydberg constant
-``m_e``                      electron mass
-``electron_mass``            electron mass
-``m_p``                      proton mass
-``proton_mass``              proton mass
-``m_n``                      neutron mass
-``neutron_mass``             neutron mass
+``c``                        speed of light in vacuum in m s^-1
```

Comment:
A nitpick: I think using the phrase “in” throughout this file to designate the units could be confusing, as it could easily be misread as “inch” to people taking a quick look at this. “Inch meter per second” doesn’t really make sense as a combination of units, but honestly I’ve seen worse in my career and this isn’t unimaginable. Using some punctuation mark to separate would make this unambiguous. eg:

```
vacuum: m s^-1
vacuum, m s^-1
vacuum (m s^-1)
vacuum [m s^-1]
```

Reply:
The IUPAC Green Book, 3rd edition (Quantities, Units, and Symbols in Physical Chemistry) states:

> The value of a physical quantity Q can be expressed as the product of a numerical value {Q} and
> a unit [Q]
> Q = {Q} [Q]
> Neither the name of the physical quantity, nor the symbol used to denote it, implies a particular
> choice of unit (see footnote 1, p. 4).
>
> Footnote 1: The symbol [Q] was formerly used for dimension of Q, but this symbol is used and preferred for
> unit of Q.

Therefore I would object to the following option:
> ```
> speed of light in vacuum [m s^-1]
> ```

The first two ones seems fine.

Reply:
I see your point, @scottshambaugh  and "vacuum, m s^-1" is my preferred style. Thanks to you and @jakobjakobson13 for your comments.

When I started to make that change, I realized there were more than 100 uses of "in" in the file, and then checked the [source cited for the constants](https://physics.nist.gov/cuu/Constants/Table/allascii.txt) for style. That page always uses "in" to say what units something is in. That source only uses SI units so there's no ambiguity with "inch".

Elsewhere in that same Python file, we do use inch and spell it out:

https://github.com/scipy/scipy/blob/913c3ffafd1179dccc723a23fca3f0d661f633dc/scipy/constants/__init__.py#L188

Inches are not used to define any constants, so the reader would need to assume that the constant definitions are mixing inches in with what are clearly SI units.

I think that the NIST people who compiled the first table have considered the style for how to display units more than I have, and we have a precedent for how to work around inches in the same doc (just spell it out). Therefore, I think it's best to leave it as it is.

Reply:
> the reader would need to assume that the constant definitions are mixing inches in with what are clearly SI units.

I'm dealing with a contractor right now who is reporting out moments of inertia in the units `kg in^2`, which is not even the most cursed combination I've come across in general. Will also make the point that we shouldn't expect users to need to read across other comments to find the context needed to disambiguate something confusing.

But I don't want to bikeshed too hard here, I'm fine with keeping it as-is if you think it's more coherent.

Reply:
I feel your pain! I used to teach machine design with most things in US units but some in SI. I know who uses gross units like `kg in^2` and why.

Reply:
Made this a separate column of the table.